index.js

```diff
@@ -145,7 +145,7 @@ module.exports = {
 
       let slug = req.url;
       let pathOnly = self.apos.utils.regExpQuote(slug.split('?')[0]);
-      let redirectRegEx = new RegExp(`redirect-${pathOnly}.*`);
+      let redirectRegEx = new RegExp(`redirect-${pathOnly}(\?.*)?`);
```

Comment:
This would match either:

* 1. `?` followed by anything followed by anything, or
* 2. Nothing followed by anything

To correct this it must end with `$`.

Reply:
Come to think of it, it really needs to start with `^` too, although `redirect-` in the middle would be unusual it could still happen and it greatly slows down the search not to start with `^` because the slug index cannot be used.